api/v1alpha1/scaledobject_types.go

```diff
@@ -29,6 +29,25 @@ type ScaledObject struct {
 	Status ScaledObjectStatus `json:"status,omitempty"`
 }
 
+// HealthStatus is the status for a ScaledObject's health
+type HealthStatus struct {
```

Comment:
Do you think that  we use (and extend) already existing [Conditions](https://github.com/kedacore/keda/blob/main/api/v1alpha1/condition_types.go), so we don't introduce a new similar field?

Reply:
We're intending to use the `HealthStatus` map to track the state of each individual trigger. Mapping each trigger name to a  `HealthStatus` allows us preserve the 1:1 mapping of trigger to state and enable quick look up at run time.

Conditions are an array of messages/status relating to the *ScaledObject* (not the individual triggers). 

Ultimately we don't feel that the datatype or granularity of `Conditions` is appropriate for the intent of `HealthStatus`. Additionally, there is a lot of logic around `Conditions` that we do not want to be convoluted/negatively affected by this new feature.

Reply:
Fair enough. Though I think that we should propagate this new functionality to `Conditions` somehow, maybe add another field to display the overall status. We are using `conditions[Active] to whether the ScaledObject is active, I suppose that in case of fallback, the `Active` will be still marked as true, right?

https://github.com/kedacore/keda/blob/7c3790f427721194999bd214c80967058f2992f2/api/v1alpha1/scaledobject_types.go#L19

Reply:
>We are using conditions[Active] to whether the ScaledObject is active, I suppose that in case of fallback, the Active` will be still marked as true, right?

Correct.

If we add a new field/column to display some information regarding fallback, what would that show? Fallback behaviour is at a per-trigger level rather than at a per-scaledobject level. We could potentially add something that displays if "at least one" trigger is in a fallback state, but that might confuse users if they have multiple triggers. Thoughts?

Reply:
I think that adding at least some info is better, so show that at least one trigger is in fallback state would be nice.